app/src/main/java/com/android/sample/ui/workout/WorkoutScreen.kt

```diff
@@ -0,0 +1,405 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.media3.common.MediaItem
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.ui.AspectRatioFrameLayout
+import androidx.media3.ui.PlayerView
+import com.android.sample.R
+import com.android.sample.model.workout.BodyWeightWorkout
+import com.android.sample.model.workout.Exercise
+import com.android.sample.model.workout.ExerciseDetail
+import com.android.sample.model.workout.WarmUpViewModel
+import com.android.sample.model.workout.WorkoutType
+import com.android.sample.model.workout.WorkoutViewModel
+import com.android.sample.model.workout.YogaWorkout
+import com.android.sample.ui.composables.ArrowBack
+import com.android.sample.ui.composables.CountDownTimer
+import com.android.sample.ui.composables.convertSecondsToTime
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Screen
+import kotlinx.coroutines.delay
+
+// Data class to hold the state of an exercise
+data class ExerciseState(val exercise: Exercise, var isDone: Boolean)
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun WarmUpScreenBody(
+    exerciseStateList: List<ExerciseState>?,
+    workoutName: String,
+    navigationActions: NavigationActions
+) {
+  // State variables for managing the UI and workout flow
+  var exerciseIndex by remember { mutableIntStateOf(0) }
+  val exerciseState =
+      exerciseStateList?.get(exerciseIndex)
+          ?: error("Exercise state list is null or index out of bounds")
```

Comment:
It is nice that you thought of handling the errors. Maybe it would be nice to  add a Toast such that the user can understand better what is happening under the hood.

Reply:
Solved